content/blog/jlink-to-produce-own-runtime/index.md

```diff
@@ -0,0 +1,142 @@
+instead of the full list implied by `ALL-MODULE-PATH`. You can use a
+comma-separated set of modules from the `--list-modules` output to limit it
+further if desired, for example to include the list of extras above. Doing
+so will reduce the size further, to around 66Mb, saving an extra 29Mb.
+
+As of `jdk-17+35` the full set of modules needed to do this is shown in the
+following command:
+
+`jdk-17+35/bin/jlink --add-modules java.base,java.compiler,java.datatransfer,java.desktop,java.instrument,java.logging,java.management,java.management.rmi,java.naming,java.net.http,java.prefs,java.rmi,java.scripting,java.se,java.security.jgss,java.security.sasl,java.smartcardio,java.sql,java.sql.rowset,java.transaction.xa,java.xml,java.xml.crypto,jdk.accessibility,jdk.charsets,jdk.crypto.cryptoki,jdk.crypto.ec,jdk.dynalink,jdk.httpserver,jdk.incubator.foreign,jdk.incubator.vector,jdk.internal.vm.ci,jdk.internal.vm.compiler,jdk.internal.vm.compiler.management,jdk.jdwp.agent,jdk.jfr,jdk.jsobject,jdk.localedata,jdk.management,jdk.management.agent,jdk.management.jfr,jdk.naming.dns,jdk.naming.rmi,jdk.net,jdk.nio.mapmode,jdk.sctp,jdk.security.auth,jdk.security.jgss,jdk.unsupported,jdk.xml.dom,jdk.zipfs --output jre-17+35 --strip-debug --no-man-pages --no-header-files --compress=2`
+
+## Can I make my runtime even smaller?
+
+As mentioned in the introduction, yes you can!  You just need to identify which modules your
+application requires.  You can do this with the
+[jdeps](https://docs.oracle.com/en/java/javase/17/docs/specs/man/jdeps.html)
+command.  For example if I take a simple "Hello World!" application I can
+get the list of modules it requires as follows:
+
+```
+$ jdk-17+35/bin/jdeps Hello.class
+Hello.class -> java.base
+   <unnamed>        -> java.io        java.base
+   <unnamed>        -> java.lang      java.base
+```
+
+This first line shows that my simple application only requires the
+`java.base` module.  In a more complicated example there will be several of
+these present in the output.  The rest of the output is a breakdown of what
+packages my application uses and which modules they relate to.  I can
+therefore create an even smaller custom runtime that will be suitable for my
+application using the following `jlink` command:
+
+```
+   jdk-17+35/bin/jlink --add-modules java.base --output jdk-17+35-minimaljre \
+      --strip-debug --no-man-pages --no-header-files --compress=2
+```
+
+And the new runtime, which comes in at only 32Mb, will still be able to run
+my "Hello World" application.  From this you can see that using jlink you
+can generate custom runtimes for your application which are even smaller
+than any JRE we could ship to you.
+
+The `jdeps` command can also be given a jar file instead of a class and show
+you all of the modules your packaged application requires.
+
+**WARNING:** In some cases the cryptography modules may not get picked up by
```

Comment:
Consider changing this "WARNING" to "NOTE:"

Reply:
Yeah, I don't mind either way, so I'll change it :-)